src/robot/model/fixture.py

```diff
@@ -15,10 +15,16 @@
 
 from collections.abc import Mapping
 
+from typing import Type, TypeVar, TYPE_CHECKING, Union
+if TYPE_CHECKING:
+    from robot.running import model
+    from robot import result
+    from robot.model import keyword
+    F = TypeVar("F", keyword.Keyword, result.Keyword, model.Keyword)
 
-def create_fixture(fixture, parent, fixture_type):
+def create_fixture(fixture: Union["F", None], parent: object, fixture_type: str) -> "F":
     # TestCase and TestSuite have 'fixture_class', Keyword doesn't.
-    fixture_class = getattr(parent, 'fixture_class', parent.__class__)
+    fixture_class: Type["F"] = getattr(parent, 'fixture_class', parent.__class__)
```

Comment:
This looks _really_ complicated for a method that's only used internally. If this was to be typed, `@overload` would probably help, but I believe it's not worth the effort.